MBBSEmu/HostProcess/ExportedModules/ExportedModuleBase.cs

```diff
@@ -668,56 +671,42 @@ private protected ReadOnlySpan<byte> StringFromArray(ReadOnlySpan<byte> inputArr
         /// <returns></returns>
         private protected ReadOnlySpan<byte> ProcessTextVariables(ReadOnlySpan<byte> outputBuffer)
         {
-            using var newOutputBuffer = new MemoryStream(outputBuffer.Length);
-            for (var i = 0; i < outputBuffer.Length; i++)
-            {
-                //Look for initial signature byte -- faster
-                if (outputBuffer[i] != 0x1)
-                {
-                    newOutputBuffer.WriteByte(outputBuffer[i]);
-                    continue;
-                }
+            //Bypass if no variable if even found
+            if (!_textVariableService.HasVariable(outputBuffer))
+                return outputBuffer;
 
-                //If we found a 0x1 -- but it'd take us past the end of the buffer, we're done
-                if (i + 3 >= outputBuffer.Length)
-                    break;
+            var txtvarsFound = _textVariableService.ExtractVariableDefinitions(outputBuffer);
 
-                //Look for full signature of 0x1,0x4E,0x26
-                if (outputBuffer[i + 1] != 0x4E && outputBuffer[i + 2] != 0x26)
-                    continue;
+            var txtvarDictionary = new Dictionary<string, string>();
 
-                //Increment 3 Bytes
-                i += 3;
+            //Get Their Values
+            var txtvarMemoryBase = Module.Memory.GetVariablePointer("TXTVARS");
```

Comment:
why are we storing txtvars in module memory? I thought when something needs to be resolved it just calls the registered function pointer?

Reply:
Yeah -- `TXTVARS` is an exported array of `TXTVAR` structs which contain the variable name and pointer for the routine to execute to get the value.